### PR TITLE
r/storage_class - add support for `allowed_topologies` + some validations

### LINKED ISSUE #391: Feature: allowed_topologies field in kubernetes storage class
### Terraform Version
Terraform v0.11.11

### Kubernetes Version
Kubernetes 1.12.7, eks.1

### Affected Resource(s)
- kubernetes_storage_class

### Important Factoids
The `zone` and `zones` parameters have been deprecated as of kubernetes 1.13, and have been replaced with [allowedTopologies](https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies)

### Example
```
resource "kubernetes_storage_class" "example" {
  metadata {
    name = "terraform-example"
  }
  storage_provisioner = "kubernetes.io/aws-ebs"
  volume_binding_mode = "WaitForFirstConsumer"
  reclaim_policy = "Retain"
  parameters {
    type = "gp2"
  }
  allowed_topologies {
    match_label_expressions {
      key = "failure-domain.beta.kubernetes.io/zone"
      values = [
        "us-east-1a",
        "us-east-1b"
      ]
    }
  }
}
```

### PR DESCRIPTION
**Pull request by DrFaust92**
### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch? (If so, please include the test log in a gist)

```
--- PASS: TestAccKubernetesStorageClass_basic (23.41s)
--- PASS: TestAccKubernetesStorageClass_importBasic (6.34s)
--- PASS: TestAccKubernetesStorageClass_generatedName (6.52s)
--- PASS: TestAccKubernetesStorageClass_importGeneratedName (5.46s)
--- PASS: TestAccKubernetesStorageClass_allowedTopologies (4.11s)
--- PASS: TestAccKubernetesDataSourceStorageClass_basic (2.91s)
```

### References
Closes #391, #891

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: kubernetes/resource_kubernetes_storage_class.go [hunk 1]
```diff
@@ -42,12 +44,21 @@ func resourceKubernetesStorageClass() *schema.Resource {
 				Description: "Indicates the type of the reclaim policy",
 				Optional:    true,
 				Default:     "Delete",
+				ValidateFunc: validation.StringInSlice([]string{
+					"Recycle",
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply kubernetes/resource_kubernetes_storage_class.go]: done

File: kubernetes/resource_kubernetes_storage_class_test.go [hunk 2]
```diff
@@ -184,6 +187,31 @@ func TestAccKubernetesStorageClass_importGeneratedName(t *testing.T) {
...
+	var conf api.StorageClass
+	resourceName := "kubernetes_storage_class.test"
+	name := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckKubernetesStorageClassDestroy,
+		Steps: []resource.TestStep{
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply kubernetes/resource_kubernetes_storage_class_test.go:169]: its relevant for AWS as well.

[the remark this answers is not in the document]
Comment (alexsomesan) [reply kubernetes/resource_kubernetes_storage_class_test.go:168]: Would the test run on AWS as is (with `storage_provisioner    = "kubernetes.io/gce-pd")?
If not, you might want to insert the value of storage_provisioner as a function parameter to the config function and determine it based on runtime environment of the test.

**Review by alexsomesan — COMMENTED** [review]
Looks good. One last thing to clarify around the test configuration (see observation by @dak1n1 )

File: kubernetes/resource_kubernetes_storage_class_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dak1n1) [reply kubernetes/resource_kubernetes_storage_class_test.go:168]: I actually did a bunch of work in this file on Friday. I'm thinking I'll cherry-pick the changes into my branch here.

https://github.com/hashicorp/terraform-provider-kubernetes/pull/1027/commits/fe827fb030de34745e6028a3ad632d7bdef38aac

**Comment by dak1n1** [comment]
I got the changes into my PR here https://github.com/hashicorp/terraform-provider-kubernetes/pull/1027/commits/fe827fb030de34745e6028a3ad632d7bdef38aac

The acceptance test is passing. Thanks for this feature!

**Pull request closed without merging**